### PR TITLE
Service for mapping public part of the certificate to the MF userid

### PR DESCRIPTION
**Pull request by balhar-jakub**
Copying the information

We have opened related PRs:
Documentation and setup - zowe/docs-site#1443
Testing - zowe/zowe-install-packaging#1713
Set up for the CICD pipeline: zowe/zowe-install-packaging#1714 and zowe/zowe-install-packaging#1726

As for the API ML the functionality is merged and hidden behind a feature flag.

Signed-off-by: Jakub Balhar <jakub.balhar@broadcom.com>

### CODE REVIEW

**Review by ifakhrutdinov — CHANGES_REQUESTED** [review]
Please fix the overlay and consider splitting return codes.

File: c/certificateService.c [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+    short distinguishedNameLength;
+    char distinguishedName[246];
+    short registryNameLength;
+    char registryName[255];
+    int certificateLength;
+    char certificate[4096];
+} RUsermapParamList;
+
+static void setValidResponseCode(HttpResponse *response, int rc, int returnCode, int returnCodeRacf, int reasonCodeRacf) {
+  if (rc == SUCCESS_RC && returnCode == SUCCESS_RC && returnCodeRacf == SUCCESS_RC && reasonCodeRacf == SUCCESS_RC) {
```
[the remark this answers is not in the document]
Comment (balhar-jakub) [reply c/certificateService.c]: I changed the codes, I am only keeping the rc if you don't mind as it is part of the API and we expect it. I would have to remove it from other repositories as well.

**Review by ifakhrutdinov — APPROVED** [review]
LGTM, but I haven't tested this. @1000TurquoisePogs can someone verify this functionality?

**Comment by 1000TurquoisePogs** [comment]
https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/zss/1.16.0-PR-224/zss-1.16.0-pr-224-4-20201014103628.pax is the build that was generated from this, and we can put it into the zowe-install* PRs as a way to test.

**Comment by 1000TurquoisePogs** [comment]
https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/zss/1.16.0-PR-224/zss-1.16.0-pr-224-4-20201014103628.pax is the build that was generated from this, and we can put it into the zowe-install* PRs as a way to test.

**Comment by 1000TurquoisePogs** [comment]
https://zowe.jfrog.io/zowe/libs-snapshot-local/org/zowe/zss/1.16.0-PR-224/zss-1.16.0-pr-224-4-20201014103628.pax is the build that was generated from this, and we can put it into the zowe-install* PRs as a way to test.

**Pull request merged**